pkg/controller/bastion/options.go

```diff
@@ -0,0 +1,184 @@
+		DiskName:            DiskResourceName(baseResourceName),
+		Location:            cluster.Shoot.Spec.Region,
+		ResourceGroupName:   cluster.ObjectMeta.Name,
+		NicName:             NicResourceName(baseResourceName),
+		Tags:                tags,
+		SecurityGroupName:   NSGName(clusterName),
+	}, nil
+}
+
+func generateBastionBaseResourceName(clusterName string, bastionName string) (string, error) {
+	if clusterName == "" {
+		return "", fmt.Errorf("clusterName can't be empty")
+	}
+	if bastionName == "" {
+		return "", fmt.Errorf("bastionName can't be empty")
+	}
+
+	staticName := clusterName + "-" + bastionName
+	h := sha256.New()
+	_, err := h.Write([]byte(staticName))
+	if err != nil {
+		return "", err
+	}
+	hash := fmt.Sprintf("%x", h.Sum(nil))
+	if len([]rune(staticName)) > maxLengthForBaseName {
+		staticName = staticName[:maxLengthForBaseName]
+	}
+	return fmt.Sprintf("%s-bastion-%s", staticName, hash[:5]), nil
+}
+
+func ingressPermissions(bastion *extensionsv1alpha1.Bastion) ([]string, error) {
+	var cidrs []string
+	for _, ingress := range bastion.Spec.Ingress {
+		cidr := ingress.IPBlock.CIDR
+		ip, ipNet, err := net.ParseCIDR(cidr)
+		if err != nil {
+			return nil, fmt.Errorf("invalid ingress CIDR %q: %w", cidr, err)
+		}
+
+		normalisedCIDR := ipNet.String()
+
+		if ip.To4() != nil {
+			cidrs = append(cidrs, normalisedCIDR)
+		} else if ip.To16() != nil {
+			// Only IPv4 is supported in sourceRanges[].
+			// https://cloud.google.com/compute/docs/reference/rest/v1/firewalls/insert
+			return nil, fmt.Errorf("IPv6 is currently not fully supported: %w", err)
+		}
+
+	}
+
+	return cidrs, nil
+}
+
+// NodesResourceName is subnetwork resource name
+func NodesResourceName(baseName string) string {
+	return fmt.Sprintf("%s-nodes", baseName)
+}
+
+// PublicIPResourceName is Public IP resource name
+func PublicIPResourceName(baseName string) string {
+	return fmt.Sprintf("%s-public-ip", baseName)
+}
+
+// NSGIngressAllowSSHResourceName is network security group ingress allow ssh  rule name
+func NSGIngressAllowSSHResourceName(baseName string) string {
+	return fmt.Sprintf("%s-allow-ssh", baseName)
+}
+
+// NSGEgressAllowOnlyResourceName is network security group egress allow only rule name
+func NSGEgressAllowOnlyResourceName(baseName string) string {
+	return fmt.Sprintf("%s-egress-worker", baseName)
+}
+
+// NSGEgressDenyAllResourceName is network security group egress deny all rule name
+func NSGEgressDenyAllResourceName(baseName string) string {
+	return fmt.Sprintf("%s-deny-all", baseName)
+}
+
+// NSGName is network security group resource name
+func NSGName(baseName string) string {
+	return fmt.Sprintf("%s-workers", baseName)
+}
+
+// DiskResourceName is Disk resource name
+func DiskResourceName(baseName string) string {
+	return fmt.Sprintf("%s-disk", baseName)
+}
+
+// NicResourceName is Nic resource name
+func NicResourceName(baseName string) string {
+	return fmt.Sprintf("%s-nic", baseName)
+}
```

Comment:
Why are these functions exported? As far as I can see they are only used in this package and in the tests.

Reply:
@dkistner the package `bastion_test` in `test/integration/bastion/bastion_test.go`  which I refer from `test/integration/infrastructure/infrastructure_suite_test.go`. that is why needs exported from `package bastion`. I'm not sure if name standard in here I need to follow.  I'm wondering whether I can change package `bastion_test` to `package bastion` in `test/integration/bastion/bastion_test.go`. in this way these functions can not exported